enterprise/server/remote_execution/uffd/uffd.go

```diff
@@ -268,11 +268,6 @@ func resolvePageFault(uffd uintptr, faultingRegion uint64, src uint64, size uint
 	}
 	_, _, errno := syscall.Syscall(syscall.SYS_IOCTL, uffd, UFFDIO_COPY, uintptr(unsafe.Pointer(&copyData)))
 	if errno != 0 {
-		if errno == unix.ENOENT {
-			// The faulting process changed its virtual memory layout simultaneously with an outstanding UFFDIO_COPY
-			// The UFFDIO_COPY will have aborted and the handler should continue to serve page faults
-			return 0, nil
-		}
```

Comment:
i'm not sure we want to remove this? 

https://man7.org/linux/man-pages/man2/ioctl_userfaultfd.2.html

Reply:
I have a feeling that these were only happening because of the last-chunk issue, but will wait for @maggie-lou to weigh in since I think she added this check originally. In my testing it seems like these errors are unrecoverable and so we shouldn't drop them silently or else things will just hang rather than failing.

Reply:
Maybe leave a comment here, and/or catch this error and return a meaningful failure back up the stack so we know when something is really borked and what happened rather than having to puzzle it out?

Reply:
It gets caught on the following line (271) - now we treat it the same as other errors, rather than checking for ENOENT specifically

Reply:
sgtm

Reply:
Will go ahead and submit this for now - if we hit ENOENT in the future and we're sure that the right thing to do is silently drop it, we can revert this specific diff chunk.